src/engine/strat_engine/filesystem.rs

```diff
@@ -47,11 +55,29 @@ impl StratFilesystem {
         }
     }
 
-    pub fn check(&self, dm: &DM) -> EngineResult<FilesystemStatus> {
+    /// check if filesystem is getting full and needs to be extended
+    /// TODO: deal with the thindev in a Fail state.
+    pub fn check(&mut self, dm: &DM) -> EngineResult<FilesystemStatus> {
         match try!(self.thin_dev.status(dm)) {
             ThinStatus::Good((_mapped, _highest)) => {
-                // TODO: check if filesystem is getting full and might need to
-                // be extended (hint: use statfs(2))
+                let mount_point = try!(self.get_mount_point());
+                let (total_bytes, total_used_bytes) = try!(fs_usage(&mount_point));
+                let free_blocks = ((*total_bytes - *total_used_bytes) / SECTOR_SIZE as u64) /
```

Comment:
slightly modified because sectors doesn't implement Div:

DataBlocks(*(total_bytes - total_used_bytes).sectors() /
                                             *DATA_BLOCK_SIZE);

ok with you?